config/configuration_singleton.rb

```diff
@@ -109,6 +109,25 @@ def quota_threshold
     ENV.fetch("OOD_QUOTA_THRESHOLD", 0.95).to_f
   end
 
+  # The paths to the JSON files that store the balance information
+  # Can be URL or File path. colon delimited string; though colon in URL is
+  # ignored if URL has format: scheme://path (colons preceeding // are ignored)
+  #
+  # /path/to/balance.json:https://osc.edu/balance.json
+  #
+  #
+  # @return [Array<String>] balance paths
+  def balance_paths
+    # regex uses negative lookahead to ignore : preceeding //
+    ENV.fetch("OOD_BALANCE_PATH", "").strip.split(/:(?!\/\/)/)
```

Comment:
we should probably memoize this and `quota_paths` since this is executed on every request...

```ruby
def balance_paths
  @balance_paths ||= ENV.fetch("OOD_BALANCE_PATH", "").strip.split(/:(?!\/\/)/)
end
```

Reply:
I made the change but is looking up something in the environment and doing a few operations on the value really that expensive?

Reply:
No its not. But its also easy to memoize. I guess we didn't do that before because we didn't actively memoize everything we could. Maybe its a bad idea in practice to do that there and instead utilize view caching where appropriate. For example, parsing quota files is far more expensive than getting the list of files to parse by using a regex.

Reply:
You know what adding that memoization was probably bad advice. I vaguely remember avoiding or removing excessive memoization in this file for this reason. If you want to revert you can.